src/libm/CMakeLists.txt

```diff
@@ -0,0 +1,72 @@
+# Helper executable: generates parts of the sleef header file
+add_executable(${TARGET_MKRENAME} mkrename.c)
+target_compile_options(${TARGET_MKRENAME}
+  PRIVATE ${FLAGS_BASIC_C} ${FLAGS_WALL})
+
+# Helper executable: dispatcher
+add_executable(${TARGET_MKDISP} mkdisp.c)
+target_compile_options(${TARGET_MKDISP}
+  PRIVATE ${FLAGS_BASIC_C} ${FLAGS_WALL})
+
+# TODO: set the parameters for mkrename run
+set(MKRENAME_PARAMS "")
+
+configure_file(
+  ${CMAKE_CURRENT_SOURCE_DIR}/GenerateSleefHeader.cmake.in
+  ${CMAKE_CURRENT_BINARY_DIR}/GenerateSleefHeader.cmake @ONLY)
+
+# Add sleeh header target
+add_custom_target(generate_sleef_header ALL
+  DEPENDS ${CMAKE_BINARY_DIR}/include/sleef.h)
+
+# Add custom command to process the sleef header and install it under /include
+add_custom_command(
+  OUTPUT ${CMAKE_BINARY_DIR}/include/sleef.h
+  COMMAND ${CMAKE_COMMAND} -P
+    ${CMAKE_CURRENT_BINARY_DIR}/GenerateSleefHeader.cmake
+  DEPENDS
+    ${CMAKE_CURRENT_SOURCE_DIR}/sleeflibm.h.org
+    ${CMAKE_CURRENT_BINARY_DIR}/GenerateSleefHeader.cmake)
+
+# Install header files in the include directory
+add_custom_target(install_headers ALL
+  DEPENDS ${CMAKE_BINARY_DIR}/include/rename.h)
+
+# Add custom command to install the rename header under /include
+add_custom_command(
+  OUTPUT ${CMAKE_BINARY_DIR}/include/rename.h
+  COMMAND ${CMAKE_COMMAND} -E copy
+    ${CMAKE_CURRENT_SOURCE_DIR}/rename.h ${CMAKE_BINARY_DIR}/include/rename.h
+  DEPENDS ${CMAKE_CURRENT_SOURCE_DIR}/rename.h)
+
+# Build main library
+add_library(${TARGET_LIBSLEEF}
+  SHARED sleefdp.c sleefsp.c ${CMAKE_BINARY_DIR}/include/sleef.h ${CMAKE_BINARY_DIR}/include/rename.h)
```

Comment:
I think that `rename.h` should not be listed as a source file but via an include path, as it is used as follows:

```
#ifdef DORENAME
#include "rename.h"
#endif
```